frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.tsx

```diff
@@ -10,31 +10,44 @@ import { WidthBoundTableInfo } from "./TableInfoPopover.styled";
 
 export const POPOVER_DELAY: [number, number] = [500, 300];
 
+interface TableSubset {
+  id: number | string;
+  description?: string;
+}
+
 const propTypes = {
-  tableId: PropTypes.number.isRequired,
+  table: PropTypes.shape({
+    id: PropTypes.oneOf([PropTypes.number, PropTypes.string]),
+    description: PropTypes.string,
+  }).isRequired,
   children: PropTypes.node,
   placement: PropTypes.string,
   offset: PropTypes.arrayOf(PropTypes.number),
 };
 
-type Props = { tableId: number } & Pick<
+type Props = { table: TableSubset } & Pick<
   ITippyPopoverProps,
   "children" | "placement" | "offset"
 >;
 
 const className = "table-info-popover";
 
-function TableInfoPopover({ tableId, children, placement, offset }: Props) {
+function TableInfoPopover({ table, children, placement, offset }: Props) {
   placement = placement || "left-start";
 
-  return tableId != null ? (
+  const { id, description } = table;
+  const hasDescription = !!description;
+  const isVirtualTable = typeof id === "string";
```

Comment:
You can use`isVirtualCardId` function from `frontend/src/metabase/lib/saved-questions/saved-questions.js`